actors/migration/nv7/top.go

```diff
@@ -48,6 +48,38 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, stateRootIn cid
 	if err != nil {
 		return cid.Undef, err
 	}
+	stateRootOut, err := adt.MakeEmptyMap(adtStore).Root()
+	if err != nil {
+		return cid.Undef, err
+	}
+	actorsOut, err := states.LoadTree(adtStore, stateRootOut)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	err = actorsIn.ForEach(func(addr address.Address, actorIn *states.Actor) error {
+		if !actorIn.Code.Equals(builtin.StorageMinerActorCodeID) { // non miners are written with no change
+			return actorsOut.SetActor(addr, actorIn)
+		}
+		minerResult, err := MinerMigrator{}.MigrateState(ctx, store, actorIn.Head, MigrationInfo{
+			address:    addr,
+			balance:    actorIn.Balance,
+			priorEpoch: priorEpoch,
+		})
+		if err != nil {
+			return err
+		}
+		minerActorOut := states.Actor{
+			Code:       builtin.StorageMinerActorCodeID,
+			Head:       minerResult.NewHead,
+			CallSeqNum: actorIn.CallSeqNum,
+			Balance:    actorIn.Balance,
```

Comment:
Nit: It's probably worth us establishing some more re-usable pieces here to aid future migrations. This ignores the transfer from the minerResult, which we "know" is zero right now, but is error-prone to bake in. So something that does this generically from a `StateMigrationResult`.